server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/ImapStepdefs.java

```diff
@@ -62,6 +62,18 @@ public void hasMessageInMailbox(String mailbox) throws Throwable {
         }
     }
 
+    @Then("^the message is \"([^\"]*)\" on \"([^\"]*)\"$")
+    public void hasMessageWithFlagInMailbox(String flags, String mailbox) throws Throwable {
+        try (IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST, IMAP_PORT);) {
+            assertThat(
+                imapMessageReader.userReceivedMessageWithFlagsInMailbox(userStepdefs.getConnectedUser(),
```

Comment:
You use ``userStepdefs.getConnectedUser()`` twice I would extract that to a var.
I would also add a "stateless" version of your state "^the message is \"([^\"]*)\" on \"([^\"]*)\" for  \"([^\"]*)\"$" with "hasMessageWithFlagInMailbox(String flags, String mailbox, String user)"